### PR TITLE
Fix ref errors and z index

### PR DESCRIPTION
**Pull request by dominik-zeglen**
This PR fixes:
1. warning that ref is not properly mounted (it's a common problem with downshift and portals)
2. z-index in popper, because autocomplete components were unusable inside dialogs

### CODE REVIEW

File: src/Autocomplete/styles.ts [hunk 1]
```diff
@@ -11,6 +11,8 @@ const useStyles = makeStyles(
     },
     popper: {
       marginTop: theme.spacing(1),
+      // Places popper above dialogs
+      zIndex: 1301,
```
Comment (orzechdev) [inline src/Autocomplete/styles.ts:15]: Shouldn't we add somewhere in the config or theme file something like the following and reference it everywhere?
```ts
zIndexConfig = {
  dialog: 1300,
  popper: 1301,
  whatever: 4200
}
```

Comment (orzechdev) [reply src/Autocomplete/styles.ts:15]: and obviously in css reference like
```ts
zIndex: theme.zIndexConfig.popper,
```

Comment (dominik-zeglen) [reply src/Autocomplete/styles.ts:15]: That's great idea actually

**Review by orzechdev — APPROVED** [review]

**Review by taniotanio7 — COMMENTED** [review]
I've changed import for `CircularProgress` in [src/Autocomplete/Autocomplete.tsx](https://github.com/saleor/macaw-ui/pull/138/files#diff-601b83d81d39ea5375e250bcb536a3c79c394b25edb7e56769130c05ae211011)

because ESLint was throwing errors. Maybe we should get rid of that ESLint rule and change every import from Mui to `@material-ui/core`?


I have also another **one fix to rule them all** proposition for z-indexes:
- Add `isolation: isolate` to the div we mount our app in to create new stacking context
- Add `isolation: isolate` to every Portal
- Remove z-indexes from Portals

This way we create a new stacking context for our app and every single Portal and if new portal appears it will cover the previous one and we don't have to worry about updating z-indexes anymore 🥳

Before (removed z-indexes):
CleanShot 2022-04-27 at 10 16 54@2x

After:
CleanShot 2022-04-27 at 10 17 03@2x

CleanShot 2022-04-27 at 10 17 10@2x

**Review by taniotanio7 — APPROVED** [review]
Added PR in dashboard to add `isolation: isolate`

https://github.com/saleor/saleor-dashboard/pull/2024

**Pull request merged**